rewrite-pattern.js

```diff
@@ -728,9 +809,51 @@ const rewritePattern = (pattern, flags, options) => {
 	};
 
 	const tree = parse(pattern, flags, regjsparserFeatures);
+
+	if (config.transform.modifiers) {
+		const allDisabledModifiers = Object.create(null)
+		const itemStack = [tree];
+		let node;
+		while (node = itemStack.pop(), node != undefined) {
```

Comment:
Instead of doing a full traversal of the AST, can we do it at the same time as the transform traversal?

i.e. `config.modifiersData.removeIfTopLevel = { i: undefined, u: undefined, s: undefined }`, and when we find a `(?-i:...)` modifier we do `config.modifiersData.removeIfTopLevel.i !== true`, without ever resetting it to false.

Then, we only generate the new regexp flags at the end, by filtering the initial flags removing all the ones set to `true` in `removeIfTopLevel`.

Reply:
To be honest, I don't like this approach either, but I didn't think of other suitable methods.
`/a((((((((?-i:b)))))))/i`
For example, in this case, when we traverse to `?-i:b`, it is already late, and `a` needs to be reprocessed.

Reply:
Unless we treat `a` as `aA` forever, but that introduces a lot of unnecessary size increase and performance hit.

Reply:
Oh I see. Maybe we could at least see the traversal if `/\(\?[a-z-]/.test(pattern)` is false.

Reply:
I don't quite get what you mean, but I tried something similar instead of traversing, not sure if there's a hidden problem, but it looks fine to me personally.